src/main/scala/com/advancedtelematic/campaigner/http/UpdateResource.scala

```diff
@@ -72,14 +74,22 @@ class UpdateResource(extractNamespace: Directive1[Namespace], deviceRegistry: De
     }
   }
 
-  private def getUpdates(ns: Namespace, offset: Long, limit: Long, sortBy: Option[SortBy], nameContains: Option[String]) =
+  private def availableUpdatesForGroups(ns: Namespace, groups: Set[GroupId]): Route = {
+    onComplete(getGroupUpdates(ns, groups)) {
+      case Success(r) => complete(r)
+      case Failure(_: TimeoutException) => complete((StatusCodes.EnhanceYourCalm, "It took too long to retrieve all the devices in the given groups."))
```

Comment:
I would prefer to return `504 Gateway Timeout` or `500 Internal Server Error` here with an appropriate error object as a payload. 4xx means processing failed because of clients error. It is definitely not a case here

Reply:
Sure, updated.